clients/include/testing_gebsr2csr.hpp

```diff
@@ -0,0 +1,626 @@
+                                 bsr_descr,
+                                 bsr_val,
+                                 bsr_row_ptr,
+                                 bsr_col_ind,
+                                 row_block_dim,
+                                 col_block_dim,
+                                 csr_descr,
+                                 csr_val,
+                                 csr_row_ptr,
+                                 csr_col_ind);
+    verify_hipsparse_status_invalid_size(status, "Error: n is invalid");
+
+    status = hipsparseXgebsr2csr(handle,
+                                 dir,
+                                 m,
+                                 n,
+                                 bsr_descr,
+                                 bsr_val,
+                                 bsr_row_ptr,
+                                 bsr_col_ind,
+                                 -1,
+                                 col_block_dim,
+                                 csr_descr,
+                                 csr_val,
+                                 csr_row_ptr,
+                                 csr_col_ind);
+    verify_hipsparse_status_invalid_size(status, "Error: row_block_dim is invalid");
+
+    status = hipsparseXgebsr2csr(handle,
+                                 dir,
+                                 m,
+                                 n,
+                                 bsr_descr,
+                                 bsr_val,
+                                 bsr_row_ptr,
+                                 bsr_col_ind,
+                                 row_block_dim,
+                                 -1,
+                                 csr_descr,
+                                 csr_val,
+                                 csr_row_ptr,
+                                 csr_col_ind);
+    verify_hipsparse_status_invalid_size(status, "Error: col_block_dim is invalid");
+}
+
+template <typename T>
+hipsparseStatus_t testing_gebsr2csr(Arguments argus)
+{
+
+    int                  m             = argus.M;
+    int                  n             = argus.N;
+    int                  row_block_dim = argus.row_block_dim;
+    int                  col_block_dim = argus.col_block_dim;
+    hipsparseIndexBase_t csr_idx_base  = argus.idx_base;
+    hipsparseIndexBase_t bsr_idx_base  = argus.idx_base2;
+    hipsparseDirection_t dir           = argus.dirA;
+    std::string          binfile       = "";
+    std::string          filename      = "";
+    hipsparseStatus_t    status;
+
+    // When in testing mode, M == N == -99 indicates that we are testing with a real
+    // matrix from cise.ufl.edu
+    int safe_size = 100;
+    if(m == -99 && n == -99 && argus.timing == 0)
+    {
+        binfile = argus.filename;
+        m = n = safe_size;
+    }
+
+    if(argus.timing == 1)
+    {
+        filename = argus.filename;
+    }
+
+    int mb = -1;
+    int nb = -1;
+    if(row_block_dim > 0 && col_block_dim > 0)
```

Comment:
Should this be:

int mb = -1;
int nb = -1;
if(row_block_dim > 0 && col_block_dim > 0)
{
    mb = (m + row_block_dim - 1) / row_block_dim;
    nb = (n + col_block_dim - 1) / col_block_dim;
}

Reply:
No, this is to initialize mb and nb to be used to initialize a csr matrix.

Reply:
I realize but its not consistent with what we do in other routines and it is confusing be we usually associate m and n with the number of rows in columns but here it is being used as mb and nb

Reply:
'confusing' depends on its definition. I will re-write for you to be less confused.

Reply:
This is your showstopper?

Reply:
Its not a 'showstopper' but it was something I came across that I think is confusing and so I commented. Here 'm' is used more like 'mb' and then later on down below you write m= mb * row_block_dim; so 'm' is back to being what it actually is. You can achieve what you want and still have mb and m be consistent with what they actually represent and what we do in other files. If you don't want to make the change then that is fine. I just think its confusing and inconsistent so I commented. I am not forcing you to make the change.

Reply:
You have requested changed. Am good for changes, I will take in consideration your comments and make it more 'consistent' and less 'confusing'. Not a big fan of the word 'confusing' ;) it was usually used by people to say "oh this is complicated", nut here it makes sense in its right definition.

On another note, usually tests are a pain to read. I have some refactoring planned.